proposals/lip-0055.md

```diff
@@ -60,13 +61,35 @@ The separation between properties in the block header and properties in the bloc
 
 The schema for the block assets allows each module to include its serialized data individually, which makes the inclusion of module data very flexible. Each module can insert a single entry in the assets. This entry is an object containing a `moduleID` property, indicating the ID of the module handling it, and a generic `data` property that can contain arbitrary serialized data.
 
-Each entry of the block assets is then inserted in a Merkle tree, whose root is included in the block header as the `assetsRoot` property. Inserting the assets root rather than the full assets allows to bound the size of the block header while still authenticating the content of the block assets.
+Each entry of the block assets is then inserted in a Merkle tree, whose root is included in the block header as the `assetRoot` property. Inserting the assets root rather than the full assets allows to bound the size of the block header while still authenticating the content of the block assets.
+
+
+### Failing and Invalid Transactions
+
+In the current protocol, a transaction that is included in a block always implies that it was processed succesfully without any error. In particular, if an error occurs during the block generation, the transaction is simply invalid and it is not included in the block payload.
+
+In this LIP, we introduce the concept of failing transactions and cleary specify when a transaction is considered invalid.
+A transaction is *valid*, if the "transaction verification", "command verification", "before command execution", and "after command execution" stages associated to the transaction are executed successfully without errors (see the ["Block processing stages"](#block-processing-stages) section below). 
+Otherwise, a transaction is *invalid*. 
+Only valid transactions should be added to a block during the block generation, as an invalid transaction makes the whole block invalid meaning that it would be discarded by any node in the network.
+
+A valid transaction is executed successfully if additionally the "command execution" stage is executed successfully without errors. 
+A valid transaction *fails* if  on the other hand an error occurs during the command execution. 
+In this case, all state transitions of the "command execution" stage are reverted. 
+This means that the transaction has no effect except for those defined in "before command execution" and "after command execution".
+The result of the transaction execution is logged using an event emitted at the end of the "after transaction execution" stage, indicating whether the transaction was processed succesfully or an error occured.
+
+Allowing for failing transactions has the advantage that transactions that require complex and expensive "command execution" stage would still pay the fee in case of errors, as the fee is processed in the "before command execution" stage. 
```

Comment:
```suggestion
Allowing for failing transactions has the advantage that transactions that require a complex and expensive "command execution" stage would still pay the fee in case of errors, as the fee is processed in the "before command execution" stage. 
```